w3lib/url.py

```diff
@@ -531,11 +556,8 @@ def canonicalize_url(
 ) -> str:
     r"""Canonicalize the given url by applying the following procedures:
 
+    - make the URL safe (see :func:`safe_url_string`)
```

Comment:
That's actually a question I had :) It seems canonicalize_url doesn't call safe_url_string. How do we ensure it makes the URl safe in the same way? Or is it not a requirement? From the docstring I'd infer that canonicalize_url calls safe_url_string, and then does something on top of it.

Reply:
If we revert, it won't be worse than before, but the docstring will not be correct - it mentions many things, but not the whitespace handlng. So, some options:

1. Revert the docstring change
2. Write "make the URL safe" without referring to safe_url_string
3. Document that whitespace and control characters are stripped

I'm fine with all 3 :)